pkg/controllers/nodeclass/status/validation.go

```diff
@@ -0,0 +1,45 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package status
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+
+	"github.com/samber/lo"
+
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	v1 "github.com/aws/karpenter-provider-aws/pkg/apis/v1"
+)
+
+type Validation struct{}
+
+func (n Validation) Reconcile(ctx context.Context, nodeClass *v1.EC2NodeClass) (reconcile.Result, error) {
+	if _, found := lo.Find(v1.RestrictedTagPatterns, func(exp *regexp.Regexp) bool {
+		for key := range nodeClass.Spec.Tags {
+			if exp.MatchString(key) {
+				return true
+			}
+		}
+		return false
+	}); found {
+		nodeClass.StatusConditions().SetFalse(v1.ConditionTypeValidationSucceeded, "NodeClassNotReady", "tag validation bypassed")
```

Comment:
```suggestion
		nodeClass.StatusConditions().SetFalse(v1.ConditionTypeValidationSucceeded, "TagValidationFailed", fmt.Sprintf("%q tag does not pass tag validation requirements", v))
```

nit: Typically the reason for the status condition is going to describe pretty specifically why something is in a state. In this case, the tag validation didn't succeed so I think that we should say that in the reason. The message can be even more specific and say that the validation failed because a specific tag (which we should callout by name) failed the validation